permissions/src/main/java/org/openmbee/sdvc/permissions/DefaultPermissionService.java

```diff
@@ -0,0 +1,737 @@
+        Optional<Branch> branch = branchRepo.findByProject_ProjectIdAndBranchId(projectId, branchId);
+
+        if (!branch.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Branch not found");
+        }
+
+        Branch bran = branch.get();
+        bran.setInherit(isInherit);
+        branchRepo.save(bran);
+    }
+
+    @Override
+    public void setOrgPublic(boolean isPublic, String orgId) {
+        Optional<Organization> organization = orgRepo.findByOrganizationId(orgId);
+
+        if (!organization.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Organization not found");
+        }
+
+        Organization org = organization.get();
+        org.setPublic(isPublic);
+        orgRepo.save(org);
+    }
+
+    @Override
+    public void setProjectPublic(boolean isPublic, String projectId) {
+        Optional<Project> project = projectRepo.findByProjectId(projectId);
+
+        if (!project.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Project not found");
+        }
+
+        Project proj = project.get();
+        proj.setPublic(isPublic);
+        projectRepo.save(proj);
+    }
+
+    @Override
+    public boolean hasOrgPrivilege(String privilege, String user, String orgId) {
+        Optional<Privilege> priv = privRepo.findByName(privilege);
+        if (!priv.isPresent()) {
+            throw new PermissionException(HttpStatus.BAD_REQUEST, "No such privilege");
+        }
+
+        Optional<OrgUserPerm> orgUserPerm = orgUserPermRepo.findAllByOrganization_OrganizationIdAndUser_Username(orgId, user);
+        if (!orgUserPerm.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Organization or Role not found");
+        }
+
+        Set<Privilege> privileges = orgUserPerm.get().getRole().getPrivileges();
+        if (privileges.contains(priv.get())) {
+            return true;
+        }
+
+        Optional<OrgUserPerm> orgGroupPerm = orgUserPermRepo.findAllByOrganization_OrganizationIdAndUser_Username(orgId, user);
+        if (!orgGroupPerm.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Organization or Group not found");
+        }
+
+        privileges.addAll(orgGroupPerm.get().getRole().getPrivileges());
+        return privileges.contains(priv.get());
+    }
+
+    @Override
+    public boolean hasProjectPrivilege(String privilege, String user, String projectId) {
+        Optional<Privilege> priv = privRepo.findByName(privilege);
+        if (!priv.isPresent()) {
+            throw new PermissionException(HttpStatus.BAD_REQUEST, "No such privilege");
+        }
+
+        Optional<ProjectUserPerm> projectUserPerm = projectUserPermRepo.findByProject_ProjectIdAndUser_UsernameAndInheritedIsFalse(projectId, user);
+        if (!projectUserPerm.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Organization or Role not found");
```

Comment:
it should be a set anyways, and shouldn't be throwing exception, should check groups first